### PR TITLE
Get rid from requests module

### PR DESCRIPTION
**Pull request by CyberDem0n**
It wasn't used for anything critical anyway, so it doesn't make a lot of sense to keep it as an explicit dependency.

### CODE REVIEW

File: patroni/postgresql/config.py [hunk 1]
```diff
@@ -250,6 +250,31 @@ def write_param(self, param, value):
...
+        self._store[key.lower()] = (key, value)
+
+    def __getitem__(self, key):
+        return self._store[key.lower()][1]
+
+    def __delitem__(self, key):
+        del self._store[key.lower()]
+
+    def __iter__(self):
+        return (casedkey for casedkey, mappedvalue in self._store.values())
```
Comment (sdudoladov) [inline patroni/postgresql/config.py]: `cased_key` / `mapped_value` (i.e snake_case)

**Comment by sdudoladov** [comment]
:+1:

File: patroni/postgresql/config.py (hunk 1 above)
Comment (CyberDem0n) [reply patroni/postgresql/config.py]: This is copy&paste from the `requests` module.

**Comment by CyberDem0n** [comment]
The first reason - `requests` is relatively heavy, it takes 11MB in memory on top of urllib3.
The second reason - yes, it causes some issues when the new `urllib3` is released.

Unfortunately `requests` is also dependancy for the official `kubernetes` module and for `python-consul`. With `kubernetes` is solvable. The K8s API we use is very stable, calling REST API endpoints is not a big deal, so I just implemented the part we need and we will get rid of another dependency.

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**